### DISCUSSION TITLE
bref + CDK doesn't work properly

### DISCUSSION THREAD

**Discussion by h-kenji1** [brefphp/bref#1889] [Support] [answered]
Hey, guys!

I'm trying to create a serverless Laravel API application using Lambda on CDK.
Actually, the plan was to create it using the Serverless Framework instead of CDK, but since Serverless v4 was released and is no longer free, I'm afraid that support for v3 won't be maintained for much longer. So, I decided to use CDK with Bref instead.

What I'm trying to do is to pack the laravel code in a separated directory and let lambda take care of it.
But the thing is that after deploying it, Lambda throws me the following error:

`{
  "errorMessage": "Handler `/var/task/public/index.php` must return a function or object handler. See https://bref.sh/docs/runtimes/function.html",
  "errorType": "Bref\\Runtime\\HandlerNotFound",
  "stackTrace": [
    "#0 /var/task/vendor/bref/laravel-bridge/src/HandlerResolver.php(37): Bref\\Runtime\\FileHandlerLocator->get('public/index.ph...')",
    "#1 /var/task/vendor/bref/bref/src/FunctionRuntime/Main.php(27): Bref\\LaravelBridge\\HandlerResolver->get('public/index.ph...')",
    "#2 /opt/bref/bootstrap.php(21): Bref\\FunctionRuntime\\Main::run()",
    "#3 {main}"
  ]
}`

If I go to index/public.php and just return a simple string it works, but every routing or anything I try to access on the Laravel side ends up not working. Any idea on how to fix it? 

Here's a sample of my CDK code:
    const brefRuntimeLayerArn = "arn:aws:lambda:ap-northeast-1:534081306603:layer:php-83:41"
      const brefLayer = LayerVersion.fromLayerVersionArn(this, "php-83", brefRuntimeLayerArn)
      const appFunction = new PhpFpmFunction(this, 'backend-service', {
        code: packagePhpCode(`${laravelProjectDir}`, {
             exclude: ['docs'],
         }),
        phpVersion: '8.3',
        handler: 'public/index.php',
        memorySize: 2048,
        reservedConcurrentExecutions: 1,
        timeout: Duration.seconds(20),
        layers: [brefLayer],
    });

**Comment by mnapoli** [comment, marked as answer]
The layer is incorrect, you're deploying a FPM function with a non-FPM layer.

**Reply by h-kenji1** [reply]
Thanks for the help. Really appreciate it.
Actually, just removing the layer from the code solved the problem. 
I tried to use a fpm layer instead like you said, but Lambda threw an error saying that it's not possible to use two versions of the same layer on the same function.
And then I realized that if you use PhpFpmFunction the layers are automatically added so it's not necessary to declare it again, and that was the main cause of the error I suppose.
Thank you for the help.
(Also, in case someone is struggling with it like I was, I'm gonna add some more information on the post above)
